File: XUCore/additions/NSShadowAdditions.m

//
// NSShadowAdditions.m
// 
// Created by Charlie Monroe
// 
// Copyright (c) 2010-13 Fuel Collective, LLC. All rights reserved.
// 


#import "NSShadowAdditions.h"

@implementation NSShadow (NSShadowAdditions)
+(NSShadow*)shadow{
	return [self shadowWithColor:[NSColor clearColor] offset:NSZeroSize blurRadius:0.0];
}
+(NSShadow*)shadowWithColor:(NSColor*)color offset:(NSSize)offset blurRadius:(CGFloat)radius{
	NSShadow *shadow = [[NSShadow alloc] init];
	[shadow setShadowColor:color];
	[shadow setShadowOffset:offset];
	[shadow setShadowBlurRadius:radius];
	return shadow;
}

@end

File: XUCore/additions/NSShadowAdditions.swift
//
//  NSShadowAdditions.swift
//  XUCore
//
//  Created by Charlie Monroe on 11/10/15.
//  Copyright © 2015 Charlie Monroe Software. All rights reserved.
//

#if os(macOS)
	import AppKit
#else
	import UIKit
#endif

public extension NSShadow {
	
	public convenience init(color: NSColor, offset: CGSize, blurRadius: CGFloat) {
		self.init()
		
		self.shadowColor = color
		self.shadowOffset = offset
		self.shadowBlurRadius = blurRadius
	}
	
}
